M2/Macaulay2/packages/Tableaux/documentation.m2

```diff
@@ -0,0 +1,1193 @@
+      an additional amount to shift.
+Outputs
+    S:SkewTableau
+      each row i of T has been shifted by i + n to the right (where n=0 if not provided).
+Description
+  Example
+    T = skewTableau(new Partition from {6,6,5,3,1})
+    shift T
+    shift(T,2)
+SeeAlso
+  unshift
+///
+
+doc ///
+Key
+    unshift
+   (unshift, SkewTableau)
+   (unshift, SkewTableau, ZZ)
+Headline
+   unshift a tableau
+Usage
+   unshift T
+   unshift(T,n)
+Inputs
+    T:SkewTableau
+      a skew tableau.
+    n:ZZ
+      an additional amount to shift.
+Outputs
+    S:SkewTableau
+      each row i of T has been shifted by i + n to the left (where n=0 if not provided).
+Description
+  Example
+    T = skewTableau(new Partition from {6,6,5,3,1})
+    T' = shift T
+    unshift T'
+SeeAlso
+  shift
+///
+
+doc ///
+Key
+   (conjugate, SkewTableau)
+Headline
+   conjugate a tableau
+Usage
+   conjugate T
+Inputs
+    T:SkewTableau
+      a tableau.
+Outputs
+    S:SkewTableau
+      the rows and columns have been switched.
+Description
+  Text
+    This method conjugates the tableau only if the shape has weakly decreasing parts, and no negative parts.
+    Otherwise, it will raise an error.
+  Example
+    T = skewTableau(new Partition from {6,6,5,3,1}, new Partition from {2,1,1}, toList(1..17))
+    conjugate T
+///
+
+doc ///
+Key
+    hookLength
+   (hookLength, Sequence, SkewTableau)
+Headline
+   compute the hook length of a box of a tableau
+Usage
+   hookLength(thePosition,T)
+Inputs
+    thePosition:Sequence
+      a pair (i,j) of integers.
+    T:SkewTableau
+      a tableau.
+Outputs
+    n:ZZ
+      the hook length of box (i,j) in T.
+Description
+  Text
+    The hook length of box (i,j) is defined as the number of boxes directly below, and directly to the right,
+    of a box, including the box itself.
+  Example
+    T = skewTableau(new Partition from {6,6,5,3,1}, new Partition from {2,1,1})
+    hookLength((1,1),T)
+    applyPositions(T, thePosition -> hookLength(thePosition,T))
+///
+
+doc ///
+Key
+    boxContent
+   (boxContent, ZZ, ZZ)
+Headline
+   compute the content of a box of a tableau
+Usage
+   boxContent(i,j)
+Inputs
+    i:ZZ
+      the row index of a box.
+    j:ZZ
+      the column index of a box.
+Outputs
+    n:ZZ
+      the hook length of box (i,j).
+Description
+  Text
+    The content of box (i,j) is defined as $j-i$.
+  Example
+    T = skewTableau(new Partition from {6,6,5,3,1}, new Partition from {2,1,1})
+    boxContent(1,1)
+    applyPositions(T, thePosition -> boxContent thePosition)
+///
+
+doc ///
+Key
+    isSkew
+   (isSkew, SkewTableau)
+Headline
+   check if a tableau is skew
+Usage
+   isSkew T
+Inputs
+    T:SkewTableau
+      a tableau.
+Outputs
+    b:Boolean
+      returns false if the inner shape is $0$, and true otherwise.
+Description
+  Example
+    isSkew skewTableau(new Partition from {6,6,5,3,1}, new Partition from {2,1,1})
+    isSkew skewTableau(new Partition from {6,6,5,3,1}, new Partition from {})
```

Comment:
I'm a little confused here, likely because I don't remember the definitions, but shouldn't the output of `skewTableau` always be a skew tableau? Would it make sense to have a more general `Tableau` type as the input of `isSkew`, or rename it to `isWellDefined` instead?